app/gui/src/controller/searcher/breadcrumbs.rs

```diff
@@ -155,22 +154,22 @@ impl<'a> Builder<'a> {
     fn module_name_and_entry(
         &self,
         module: &component::Id,
-    ) -> Option<(Rc<module::QualifiedName>, BreadcrumbEntry)> {
+    ) -> Option<(Rc<QualifiedName>, BreadcrumbEntry)> {
         let module_name = self.components.module_qualified_name(*module)?;
-        let entry = BreadcrumbEntry::from(self.lookup(&module_name)?);
+        let entry = BreadcrumbEntry::from(self.lookup((*module_name).as_ref())?);
         Some((module_name, entry))
     }
 
-    fn lookup(&self, name: &module::QualifiedName) -> Option<(component::Id, Rc<Entry>)> {
-        self.database.lookup_by_qualified_name(name.into_iter())
+    fn lookup(&self, name: QualifiedNameRef) -> Option<(component::Id, Rc<Entry>)> {
```

Comment:
If this could accept `impl Into<QualifiedNameRef>`, and there were a `impl<'a> From<&'a QualifiedName> for QualifiedNameRef<'a>`, it would simplify some call sites.